android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothStandardWeightProfile.java

```diff
@@ -823,29 +825,59 @@ protected void chooseExistingScaleUser(Vector<ScaleUser> userList) {
         chooseScaleUserUi(choices);
     }
 
-    protected String getInitials(String fullName) {
-        if (fullName == null || fullName.isEmpty() || fullName.chars().allMatch(Character::isWhitespace)) {
-            return getDefaultInitials();
-        }
-        return buildInitialsStringFrom(fullName).toUpperCase();
+    protected String getInitials(final String fullName, final int size) {
+        if (0 == size)
+            return StringUtils.EMPTY;
+
+        if (StringUtils.isAnyBlank(fullName))
+            return getDefaultInitials(size);
+
+        return buildInitialsStringFrom(fullName, size).toUpperCase();
     }
 
-    private String getDefaultInitials() {
+    private String getDefaultInitials(final int size) {
+        if (1 > size)
+            return StringUtils.EMPTY;
+
         int userId = this.selectedUser.getId();
-        int userIndex = getUserScaleIndex(userId);
-        return "P" + userIndex + " ";
+        String userIndex = String.valueOf(getUserScaleIndex(userId));
+
+        StringBuilder stringBuilder = new StringBuilder();
+
+        if (userIndex.length() < size)
+            stringBuilder.append('P');
+
+        stringBuilder.append(userIndex);
+
+        final int padding = size - stringBuilder.length();
+
+        if(0 < padding)
+            stringBuilder.append(StringUtils.repeat(StringUtils.SPACE, padding));
+
+        stringBuilder.setLength(size);
+
+        return stringBuilder.toString();
     }
 
-    private String buildInitialsStringFrom(String fullName) {
-        String[] name = fullName.trim().split(" +");
-        String initials = "";
-        for (int i = 0; i < 3; i++) {
-            if (i < name.length && name[i] != "") {
-                initials += name[i].charAt(0);
-            } else {
-                initials += " ";
-            }
-        }
-        return initials;
+    private String buildInitialsStringFrom(final String fullName, final int size) {
+        String[] names = StringUtils.split(fullName);
+
+        if (null == names || 0 == names.length)
+            return StringUtils.repeat(StringUtils.SPACE, size);
+
+        StringBuilder initialsBuilder = new StringBuilder();
+
+        for (int i = 0; i < size && i < names.length; i++)
+            if (false == StringUtils.isAnyBlank(names[i]))
+                initialsBuilder.append(names[i].charAt(0));
+
+        final int padding = size - initialsBuilder.length();
+
+        if(0 < padding)
+            initialsBuilder.append(StringUtils.repeat(StringUtils.SPACE, padding));
```

Comment:
StringUtils.repeat() function accepts zero and negative values, it will generate zero-length string in such case;
```suggestion
        initialsBuilder.append(StringUtils.repeat(StringUtils.SPACE, size - initialsBuilder.length()));
```

Reply:
I changed a bit the logic on this. Still I would keep the if as it will be less calls, calculations and is also more future proof.

Reply:
My reasoning is the same as in https://github.com/krisjans/openScale/pull/4/files/28306a8b7e5ecdb7a7f3d390d26b84c8d1353508#r711721992.